### ISSUE TITLE
[master: visionfive2] mesa-pvr

### ISSUE THREAD

**Issue by jwinarske** [riscv/meta-riscv#465]
**Description**

Running `vulkaninfo` on `core-image-weston` (x11 removed) none of the extensions required to present to display are present:
```
VK_KHR_display
VK_KHR_wayland_surface
VK_KHR_xcb_surface
VK_KHR_xlib_surface
```

**Steps to reproduce the issue:**
```
MACHINE ?= "visionfive2"

INIT_MANAGER = "systemd"
DISTRO_FEATURES:append = " systemd alsa usbhost pipewire"
DISTRO_FEATURES:remove = "sysvinit usbgadget ptest xen x11"
DISTRO_FEATURES:append = " opengl wayland pam vulkan"

IMAGE_INSTALL:append = " \
    \
    curl \
    vulkan-tools \
    vulkan-caps-viewer \
    sascha-samples \
    vkcube \
    visionfive2-pvr-graphics \
    \
    libsf-codaj12 \
    libsf-omxil \
    libsf-wave420l \
    libsf-wave511 \
    jpu-module \
    vdec-module \
    venc-module \
"

PACKAGECONFIG:append:pn-mesa-pvr = " egl"
PACKAGECONFIG:remove:pn-qtbase = " gl"
PACKAGECONFIG:append:pn-qtbase = " gles2"
```

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Additional details (revisions used, host distro, etc.):**

https://github.com/jwinarske/meta-vulkan

**Comment by jwinarske** [comment]
Looks like upstream has Vulkan still as a work in progress:
https://rvspace.org/en/project/VisionFive2_Debian_Wiki_202403_Release

**Comment by jwinarske** [comment]
Upstream working reference has Vulkan display/surface support, as well as working libcamera IPU (lagging libcamera 0.2.0).  They are moving to 6.6 kernel soon with most all of their kernel drivers upstreamed:
https://github.com/starfive-tech/VisionFive2

The Pine64 STAR64 and PineTab-V use the same SoC and are based on the starfive-tech VisionFive2 repo:
https://pine64.org/devices/star64
https://pine64.org/devices/pinetab-v

If you're open to PRs I'm happy to make some updates.

**Comment by kraj** [comment]
@jwinarske yes, I am looking for updating to latest stable kernel as the upcoming yocto 5 release is a LTS one, and it will be good to have it on latest LTS of every package we can

**Comment by jwinarske** [comment]
Upstream is using `MESA3D_VERSION = 20.2.2`:
https://github.com/starfive-tech/buildroot/blob/7dd07bb55ed4f4f1507b4f2d0b1e95d39d94c0f2/package/mesa3d/mesa3d.mk

Their libcamera is behind 0.2.0 as well.

I'll take a look at re-basing mesa + libcamera to scarthgap versions.
